torch_xla/csrc/tensor.cpp

```diff
@@ -1405,6 +1405,36 @@ void XLATensor::masked_fill_(XLATensor& input, const XLATensor& mask,
                                                      expanded_mask, value));
 }
 
+XLATensor XLATensor::cross(const XLATensor& input, const XLATensor& other,
```

Comment:
I am thinking, we should have a tensor_ops.cpp/h, where all the XLA tensor ops which are a composition of other XLA tensors ops, are implemented.
So that big stuff like this can go (as well as stuff which has one use, like IndexAcrossDims()).

Reply:
Yeah, I've been thinking myself about it, `tensor.cpp` is growing fast. How far would you go? Different class? Same `XLATensor` class, but just put the methods in `tensor_ops.cpp`?

Reply:
> Yeah, I've been thinking myself about it, `tensor.cpp` is growing fast. How far would you go? Different class? Same `XLATensor` class, but just put the methods in `tensor_ops.cpp`?

So for ops that works with ir::Value we have ops.cpp/h. Anything that works with ir::Value should go there.
But we have also a few things which are a composition of other, XLA tensor based, ops.
These, together with the support APIs, should go there.
So if in tensor.cpp we have:

```
XLATensor XLATensor::foo(...) {
}
```

Which is a composition of other XLA tensor ops, we should have:

[tensor.cpp]
```
XLATensor XLATensor::foo(...) {
  return tensor_ops::foo(...);
}
```

[tensor_ops.cpp]
```
XLATensor foo(...) {
  return ...;
}
```

Reply:
Alright, free functions it is then. I'll do it for `cross` and we can convert more in subsequent patches.

Reply:
Done.